rotkehlchen/data_migrations/migrations/migration_9.py

```diff
@@ -24,7 +24,21 @@ def _reset_curve_decoded_events(write_cursor: 'DBCursor') -> None:
     The code is taken from `delete_events_by_tx_hash` right before 1.27 release and is
     modified to delete only transactions that have a decoded curve event.
     """
-    write_cursor.execute('SELECT DISTINCT tx_hash from evm_transactions LEFT JOIN history_events ON history_events.event_identifier=tx_hash WHERE counterparty IS ?', (CPT_CURVE,))  # noqa: E501
+    db_version = write_cursor.execute('SELECT value FROM settings WHERE name="version"').fetchone()
+
+    query = (
+        'SELECT DISTINCT tx_hash from evm_transactions LEFT JOIN history_events '
+        'ON history_events.event_identifier=tx_hash '
+    )
+    if db_version is None or int(db_version[0]) < 37:
+        query += 'WHERE counterparty IS ?'
```

Comment:
Data migrations happen after db upgrade so I'm not sure why this distinction here. You should assume that the schema is the latest one